apps/hash-ai-worker-ts/src/activities/infer-entities/create-entities.ts

```diff
@@ -0,0 +1,219 @@
+import type { VersionedUrl } from "@blockprotocol/type-system";
+import type { Entity, GraphApi } from "@local/hash-graph-client";
+import type { InferEntitiesCreationFailure } from "@local/hash-isomorphic-utils/temporal-types";
+import type {
+  AccountId,
+  EntityId,
+  LinkData,
+  OwnedById,
+} from "@local/hash-subgraph";
+
+import type { DereferencedEntityType } from "./dereference-entity-type";
+import type { ProposedEntitiesByType } from "./generate-functions";
+
+export const createEntities = async ({
+  actorId,
+  graphApiClient,
+  proposedEntitiesByType,
+  requestedEntityTypes,
+  ownedById,
+}: {
+  actorId: AccountId;
+  graphApiClient: GraphApi;
+  proposedEntitiesByType: ProposedEntitiesByType;
+  requestedEntityTypes: Record<
+    VersionedUrl,
+    { isLink: boolean; schema: DereferencedEntityType }
+  >;
+  ownedById: OwnedById;
+}): Promise<{
+  createdEntities: Entity[];
+  creationFailures: InferEntitiesCreationFailure[];
+}> => {
+  const nonLinkTypes = Object.values(requestedEntityTypes).filter(
```

Comment:
Might be more clear that we're not referring to property or data types here:

```suggestion
  const nonLinkEntityTypes = Object.values(requestedEntityTypes).filter(
```

Reply:
Sure – done in https://github.com/hashintel/hash/pull/3630/commits/de29e970691434cf5bd81c447811e1a14a1c8853